jenkins/clean_metadata.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Remove Jenkins agent SSH keys from a project's metadata."""
+
+from __future__ import print_function
+
+import argparse
+import json
+import requests
+import subprocess
+import sys
+import time
+
+
+AGENT_PR_SUBSTR = 'agent-pr'
+NUM_RETRIES = 6
+SLEEP_DURATION = 10
+
+
+def try_update(project, token):
+    """Try to remove agent keys. Returns True if we should retry."""
```

Comment:
Done. Go has tainted me and now I just want to return errors rather than raise exceptions.